sync/notify/bugs.py

```diff
@@ -159,34 +159,160 @@ def for_sync(sync, results):
                 continue
 
             product, component = component.split(" :: ")
-            summary, comment = bug_data(sync, test_results)
+            summary, comment = bug_data(sync,
+                                        test_results,
+                                        results.treeherder_url,
+                                        results.wpt_sha)
             bug_id = make_bug(summary, comment, product, component, [sync.bug])
             rv[bug_id] = [item + (link_status,) for item in test_results]
 
     return rv
 
 
-def bug_data_crash(sync, test_results):
-    summary = "New wpt crashes from PR %s" % sync.pr
-
-    comment = """The following tests have crashes in the CI runs for wpt PR %s:
-%s
-
-(getting the crash signature into these bug reports is a TODO; sorry)
-
-These updates will be on mozilla-central once bug %s lands.
-
-""" % (sync.pr,
-       detail_part(None, test_results, None, "head", False),
-       sync.bug)
-
-    comment += postscript
+class LengthCappedStringBuilder(object):
+    def __init__(self, max_length):
+        """Builder for a string that must not exceed a given length"""
+        self.max_length = max_length
+        self.data = []
+        self.current_length = 0
+
+    def append(self, other):
+        len_other = len(other)
+        if len_other + self.current_length > self.max_length:
+            return False
+        self.data.append(other)
+        self.current_length += len_other
+        return True
+
+    def has_capacity(self, chars):
+        return self.current_length + chars <= self.max_length
+
+    def get(self):
+        return "".join(self.data)
+
+
+def split_id(test_id):
+    parts = test_id.split("/")
+    last = None
+    for i, part in enumerate(parts):
+        if "#" in part or "?" in part:
+            last = i
+            break
+    if last:
+        name = "/".join(parts[i:])
+        parts = parts[:i]
+        parts.append(name)
+    return tuple(parts)
+
+
+def get_common_prefix(test_ids):
```

Comment:
So here python saves you a bit of time.

[`os.path.commonprefix`](https://docs.python.org/2.7/library/os.path.html#os.path.commonprefix) available in python 3 too. Takes a list

```python
>>> test_results = ["/a/b/c.html", "/a/b/d.html", "/a/b/e.html", "/a/b/f.html", "/a/b/g.html"]
>>> import os.path
>>> os.path.commonprefix(test_results)
'/a/b/'
```

That should save a bit of lines.

Reply:
And just for the pleasure.
https://github.com/python/cpython/blob/fe75b62575bcfdf1c39be71c1e50257832a596db/Lib/genericpath.py#L68-L83

Reply:
So that doesn't quite work I think. It's basically just a string prefix, so for `["a/b/cde/f.html", "a/b/cdg/h.html"]` it will return `"a/b/cd"` which doesn't seem like what we want. Obviously it's possible to fix that up but then it doesn't save so many lines. Also, in theory we always want to use `/` as the seperator whereas that would use the OS path separator. In practice this code likely only works on Linux anyway.

Reply:
Looks like it works if you pass in a list of tuples, which I suppose works for us, although it's a weird undocumented corner of the API.